cvmfs/libcvmfs_int.cc

```diff
@@ -374,6 +374,107 @@ int LibContext::ListDirectory(
   return 0;
 }
 
+int LibContext::GetNestedCatalogAttr(
+  const char *c_path,
+  struct cvmfs_nc_stat *info
+) {
+  ClientCtxGuard ctxg(geteuid(), getegid(), getpid());
+  LogCvmfs(kLogCvmfs, kLogDebug,
+    "cvmfs_stat_nested_catalog (cvmfs_nc_stat) : %s", c_path);
+
+  PathString p;
+  p.Assign(c_path, strlen(c_path));
+
+  /* Find the nested catalog from the root catalog */
+  shash::Any hash;
+  uint64_t size;
+  const bool found =
+    mount_point_->catalog_mgr()->GetRootCatalog()->FindNested(p, &hash, &size);
+  if (!found) {
+    return -ENOENT;
+  }
+
+  /* Set values of the passed structure */
+  info->mountpoint = c_path;
+  info->hash = &hash;
+  info->size = size;
+
+  return 0;
+}
+
+
+int LibContext::ListNestedCatalog(
+  const char *c_path,
+  char ***buf,
+  size_t *buflen
+) {
+  ClientCtxGuard ctxg(geteuid(), getegid(), getpid());
+  LogCvmfs(kLogCvmfs, kLogDebug,
+    "cvmfs_list_nested_file_catalog on path: %s", c_path);
+
+  if (c_path[0] == '/' && c_path[1] == '\0') {
+    // root path is expected to be "", not "/"
+    c_path = "";
+  }
+
+  PathString path;
+  path.Assign(c_path, strlen(c_path));
+
+  /* Lookup the directory entry specified */
+  catalog::DirectoryEntry dirent;
+  mount_point_->catalog_mgr()->LookupPath(path, catalog::kLookupSole, &dirent);
+
+  /* If dirent exists and is a nested catalog mountpoint, 
+   * we look past it so the catalog is mounted. */
+  if ( dirent.IsNestedCatalogMountpoint() ) {
+    /* Look at fake file */
+    const std::string fake("/.cvmfscatalog");
+    std::string extended_path = path.ToString() + fake;
+    path.Assign(extended_path.c_str(), strlen(extended_path.c_str()));
+    mount_point_->catalog_mgr()->LookupPath(path,
+                                            catalog::kLookupSole,
+                                            &dirent);
+  }
+
+  /* Find the correct catalog */
+  catalog::Catalog *found_catalog;
+  found_catalog = mount_point_->catalog_mgr()->FindCatalog(path);
+
+  size_t listlen = 0;
+  AppendStringToList(NULL, buf, &listlen, buflen);
+
+  /* Build listing */
+  catalog::Catalog *parent  = found_catalog->parent();
```

Comment:
Hm, for looking reasons, we might need to wrap the core of this call is well in the catalog manager, perhaps as `ListNestedCatalogSkein`